internal/gps/source_cache.proto

```diff
@@ -0,0 +1,32 @@
+syntax = "proto3";
+package gps;
+
+// ConstraintMsg is a serializable representation of a Constraint or UnpairedVersion.
+message ConstraintMsg {
+	enum Type {
+		Revision = 0;
+		Branch = 1;
+		DefaultBranch = 2;
+		Version = 3;
+		Semver = 4;
+	}
+	Type type = 1;
+	string value = 2;
+	//TODO strongly typed Semver field
```

Comment:
We're set up to drop whatever sort of Semver message type we come up with into here (and in a forward compatible way, if we want to commit first).

Reply:
It is operable - as much as the tests validate anyways.  It is using string serialization into the value field for now.